kvbc/src/resources-manager/AdaptivePruningManager.cpp

```diff
@@ -11,31 +11,51 @@ using namespace concord::performance;
 AdaptivePruningManager::AdaptivePruningManager(
     const std::shared_ptr<concord::performance::IResourceManager> &resourceManager,
     const std::chrono::duration<double, std::milli> &interval,
+    const std::shared_ptr<concordMetrics::Aggregator> &aggregator,
     concord::kvbc::IReader &ro_storage)
     : repId(bftEngine::ReplicaConfig::instance().getreplicaId()),
       resourceManager(resourceManager),
       mode(LEGACY),
       interval(interval),
-      ro_storage_(ro_storage) {
+      ro_storage_(ro_storage),
+      metricComponent(std::string("Adaptive Pruning"), aggregator),
+      ticksPerSecondMetric(metricComponent.RegisterAtomicGauge(std::string("ticks_per_second"), 0)),
+      batchSizeMetric(metricComponent.RegisterAtomicGauge("batch_size", 0)),
+      transactionsPerSecondMetric(metricComponent.RegisterAtomicGauge("transactions_per_second", 0)),
+      postExecUtilizationMetric(metricComponent.RegisterAtomicGauge("post_exec_utilization", 0)),
+      pruningAvgTimeMicroMetric(metricComponent.RegisterAtomicGauge("pruning_avg_time_micro", 0)),
+      pruningUtilizationMetric(metricComponent.RegisterAtomicGauge("pruning_utilization", 0)) {
   (void)ro_storage_;
   resourceManager->setPeriod(bftEngine::ReplicaConfig::instance().adaptivePruningIntervalPeriod);
 }
 
 AdaptivePruningManager::~AdaptivePruningManager() { stop(); }
 
-void AdaptivePruningManager::notifyReplicas(const long double &rate, const uint64_t batchSize) {
+void AdaptivePruningManager::notifyReplicas(const PruneInfo &pruneInfo) {
   if (!bftClient) {
     LOG_ERROR(ADPTV_PRUNING, "BFT client is not set");
     return;
   }
   concord::messages::ReconfigurationRequest rreq;
 
   concord::messages::PruneTicksChangeRequest pruneRequest;
-
+  /*  */
   pruneRequest.sender_id = bftClient->getClientId();
-  pruneRequest.tick_period_seconds = 1;
```

Comment:
@dartdart26, does ticks per second in concurrent pruning means that n=5 ticks are generated by each replica every second? Is the frequency 1/n then?

Reply:
Ticks period means that each replica will generate a tick every `tick_period_seconds`. For example, if you configure `tick_period_seconds = 1` and there are 7 replicas, there will be 7 ticks per second in the system.